ios/browser/api/bookmarks/brave_bookmarks_api.h

```diff
@@ -0,0 +1,104 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_IOS_BROWSER_API_BOOKMARKS_BRAVE_BOOKMARKS_API_H_
+#define BRAVE_IOS_BROWSER_API_BOOKMARKS_BRAVE_BOOKMARKS_API_H_
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSUInteger, BookmarksNodeType) {
+  BookmarksNodeTypeURL,
+  BookmarksNodeTypeFOLDER,
+  BookmarksNodeTypeBOOKMARK_BAR,
+  BookmarksNodeTypeOTHER_NODE,
+  BookmarksNodeTypeMOBILE
+};
+
+typedef NS_ENUM(NSUInteger, BookmarksNodeFaviconState) {
+  BookmarksNodeFaviconStateINVALID_FAVICON,
+  BookmarksNodeFaviconStateLOADING_FAVICON,
+  BookmarksNodeFaviconStateLOADED_FAVICON,
+};
+
+@protocol BookmarkModelObserver;
+@protocol BookmarkModelListener;
+
+OBJC_EXPORT
+@interface BookmarkNode: NSObject
+@property (class, nonatomic, copy, readonly, getter=kRootNodeGuid) NSString *rootNodeGuid;
+@property (class, nonatomic, copy, readonly, getter=kBookmarkBarNodeGuid) NSString *bookmarkBarNodeGuid;
+@property (class, nonatomic, copy, readonly, getter=kOtherBookmarksNodeGuid) NSString *otherBookmarksNodeGuid;
+@property (class, nonatomic, copy, readonly, getter=kMobileBookmarksNodeGuid) NSString *mobileBookmarksNodeGuid;
+@property (class, nonatomic, copy, readonly, getter=kManagedNodeGuid) NSString *managedNodeGuid;
+
+@property (nonatomic, assign, readonly) bool isPermanentNode;
+
+@property (nonatomic, assign, readonly, getter=getNodeId) NSUInteger nodeId;
+@property (nonatomic, copy, readonly, getter=getGuid) NSString *guid;
+@property (nonatomic, nullable, copy, getter=getUrl) NSURL *url;
+@property (nonatomic, nullable, copy, readonly, getter=getIconUrl) NSURL *iconUrl;
+
+@property (nonatomic, assign, readonly, getter=getType) BookmarksNodeType type;
+@property (nonatomic, copy, getter=getDateAdded) NSDate *dateAdded;
+@property (nonatomic, copy, getter=getDateFolderModified) NSDate *dateFolderModified;
```

Comment:
Not sure why you are adding custom getters to all of these when they will auto-synthesize to the name of their property. I think you even changed some of the implementations names to these custom getters which is weird IMO